src/redux/books/books.js

```diff
@@ -1,31 +1,42 @@
+import { v4 as uuidv4 } from 'uuid';
+
 const ADD_BOOK = 'bookstore/books/addBook';
 const REMOVE_BOOK = 'bookstore/books/removeBook';
-const initialState = [];
 
-const bookReducer = (state = initialState, action) => {
-  switch (action.type) {
-    case ADD_BOOK:
-      return [...state, action.book];
-    case REMOVE_BOOK:
-      return [...state.filter((i) => i.title !== action.book.title)];
-    default:
-      return state;
-  }
+const allBooks = {
+  books: [
+    {
+      id: uuidv4(),
+      title: 'The power of Habit',
+      author: 'James Clear',
+    },
+  ],
 };
 
-const addBook = () => ({
+const addBook = (book) => ({
   type: ADD_BOOK,
-  book: {
-    title: 'the power of Habit',
-  },
+  payload: book,
 });
 
-const removeBook = () => ({
+const removeBook = (book) => ({
   type: REMOVE_BOOK,
-  book: {
-    title: 'the power of Habit',
-  },
+  book,
 });
 
+const bookReducer = (state = allBooks, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return {
+        books: [...state.books, action.payload],
+      };
+    case REMOVE_BOOK:
+      return {
+        books: [...state.books.filter((i) => i.title !== action.book.title)],
```

Comment:
- [ ] With this setup books of the same title will be filtered out of the state, but it should not be so. Your books should be filtered using the `id`. Kindly do that.